### ISSUE TITLE
Updating individual array elements always returns syntax error (v8.9.0)

### ISSUE THREAD

**Issue by Xyvyrianeth** [brianc/node-postgres#2914]
Test queries:
```sql
CREATE TABLE profiles
  (id text, elos integer[], wins integer[], loss integer[], ties integer[]);

INSERT INTO profiles
  (id, elos, wins, loss, ties)
VALUES
  ('TEST1', ARRAY[1000, 1000, 1000], ARRAY[0, 0, 0], ARRAY[0, 0, 0], ARRAY[0, 0, 0]);

INSERT INTO profiles
  (id, elos, wins, loss, ties)
VALUES
  ('TEST2', '{1000, 1000, 1000}', '{0, 0, 0}', '{0, 0, 0}', '{0, 0, 0}');

SELECT * FROM profiles;

UPDATE profiles
SET
  elos[1] = 1100,
  wins[1] = 1
WHERE id = 'TEST1';

UPDATE profiles
SET
  elos[2] = 900,
  loss[1] = 1
WHERE id = 'TEST2';

SELECT * FROM profiles;
SELECT elos[1] FROM profiles;
```

Even when queries are submitted individually, it always eventually returns
```
error: at or near "[": syntax error

  elos[1] = 1100,
      ^
```

[Here's the same set of queries working in DBFiddle](https://dbfiddle.uk/AYmZI7b4).

**Comment by brianc** [comment]
pg doesn't change your query text at all....so typically a syntax error as something you're doing wrong on your side.  That error message also doesn't look like its from pg either....can you create a self-contained file in js that demonstrates the issue?

**Comment by Xyvyrianeth** [comment]
Nothing is happening to the query text on my part; that's exactly what's put into `db.query()`. 
![image](https://user-images.githubusercontent.com/25062300/221435807-b61f5cd5-99f0-4b54-94c1-07c30a03ed23.png)
I even tried splitting the text into individual statements that get the same treatment, and only the two update statements get the error.
![image](https://user-images.githubusercontent.com/25062300/221436697-cad0fb8e-4704-4d16-919d-f475e124e929.png)
As for the error message, all of what's in it is from the error object that's thrown by PG, but its structure is my own.

**Comment by brianc** [comment]
> can you create a self-contained file in js that demonstrates the issue?

**Comment by brianc** [comment]
yeah I'm saying that the syntax error you're hitting is a problem _with your query's syntax_ and not the driver. The driver doesn't modify your query __in any way__ other than converting it to/from binary.

**Comment by Xyvyrianeth** [comment]
> the syntax error you're hitting is a problem _with your query's syntax_

I am telling you, that is false. If you look at the query, you will see there is nothing syntactically wrong with it, as proven by an identical query working fine in DBFiddle. If you compare it to PostgreSQL's official documentation on modifying arrays (8.15.4), you will find that syntax of the examples given precisely match the query's.

**Comment by charmander** [comment]
I can’t reproduce this issue. Are you running the pg query against a different, not-quite-PostgreSQL database? Or a very old version of PostgreSQL?

**Comment by Xyvyrianeth** [comment]
hmm that could be the issue.

I started using CockroachDB a few months ago after the service I was using to host my project before then went paid-only. I found it looking for alternatives, and it said it supported *most* PostgreSQL features, but I didn't look extensively for any parity issues at the time, and I've since forgotten I even switched to it.

Looking through their docs now, there is no information regarding updating existing array elements. I don't know if that means there's nothing different about how they handle it from how PostgreSQL handles it or if they don't handle it at all, but I've submitted a ticket asking about that information.
